### PR TITLE
HZC-960: C++ Sample Client

### PR DESCRIPTION
**Pull request by yunussandikci**
Hi, this pull request adds C++ Client documentation.
You can find a sample client from [here](https://github.com/hazelcast/hazelcast-cloud-cpp-sample-client)

### CODE REVIEW

**Review by uddr — APPROVED** [review]
lgtm

**Review by galibey — APPROVED** [review]

**Review by nosan — APPROVED** [review]

File: docs/modules/ROOT/pages/cpp-client.adoc [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+<3> Then, we set discovery token of our cluster. 
++
+This discovery token will be used by client in order to discover cluster members.
+
+<4> Now, we create the client with `config`.
+
+<5> We get/create a map with name `map`.
+
+<6> Then, we check add a value to our map, and get it to let the code throw an exception if the value is not correct.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/cpp-client.adoc:96]:
```suggestion
<6> Then, we add an entry to our map and read it back to check that everything is working.
```

File: docs/modules/ROOT/pages/cpp-client.adoc [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    return 0;
+}
+----
+
+
+<1> First, we set name of the our cluster. 
+
+<2> We enable statistics in order to see statistics in Hazelcast Cloud Console.
+
+<3> We set discovery token of our cluster.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/cpp-client.adoc:159]:
```suggestion
<3> Then, we set our cluster's discovery token. 
```

File: docs/modules/ROOT/pages/cpp-client.adoc [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+----
+
+
+<1> First, we set name of the our cluster. 
+
+<2> We enable statistics in order to see statistics in Hazelcast Cloud Console.
+
+<3> We set discovery token of our cluster. 
++
+This discovery token will be used by client in order to discover cluster members.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/cpp-client.adoc:161]:
```suggestion
The client will use this discovery token to discover cluster members.
```

File: docs/modules/ROOT/pages/cpp-client.adoc [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+<4> We set our certificate file, key file and password in order establish a secure connection with our Cluster..
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/cpp-client.adoc:163]:
```suggestion
<4> To establish a secure connection to our cluster, we set our certificate file, key file and password.
```

File: docs/modules/ROOT/pages/cpp-client.adoc [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+<5> We create the client with our `config`.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/cpp-client.adoc:165]:
```suggestion
<5> We create the client, using our `config` object.
```

File: docs/modules/ROOT/pages/cpp-client.adoc [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+<6> We get/create a map with name `map`.
+
+<7> Then, we check add a value to our map, and get it to let the code throw an exception if the value is not correct.
```
Comment (JakeSCahill) [inline docs/modules/ROOT/pages/cpp-client.adoc:169]:
```suggestion
<7> Then, we add an entry to our map and read it back to check that everything is working.
```

**Review by JakeSCahill — COMMENTED** [review]
Looks good. Just some general comments and suggestions.

File: docs/modules/ROOT/pages/cpp-client.adoc (hunk 5 above)
Comment (yunussandikci) [reply docs/modules/ROOT/pages/cpp-client.adoc]: Done, thanks!

File: docs/modules/ROOT/pages/cpp-client.adoc (hunk 6 above)
Comment (yunussandikci) [reply docs/modules/ROOT/pages/cpp-client.adoc]: Done, thanks!

File: docs/modules/ROOT/pages/cpp-client.adoc (hunk 4 above)
Comment (yunussandikci) [reply docs/modules/ROOT/pages/cpp-client.adoc]: Done, thanks!

File: docs/modules/ROOT/pages/cpp-client.adoc (hunk 3 above)
Comment (yunussandikci) [reply docs/modules/ROOT/pages/cpp-client.adoc]: Done, thanks!

File: docs/modules/ROOT/pages/cpp-client.adoc (hunk 2 above)
Comment (yunussandikci) [reply docs/modules/ROOT/pages/cpp-client.adoc]: Done, thanks!

File: docs/modules/ROOT/pages/cpp-client.adoc (hunk 1 above)
Comment (yunussandikci) [reply docs/modules/ROOT/pages/cpp-client.adoc:96]: Done, thanks!

File: docs/modules/ROOT/pages/cpp-client.adoc [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+}
+----
+
+<1> First, we set name of the our cluster. 
+
+<2> Then, we enable statistics in order to see statistics in Hazelcast Cloud Console.
+
+<3> Then, we set discovery token of our cluster. 
++
+This discovery token will be used by client in order to discover cluster members.
```
[the remark this answers is not in the document]
Comment (yunussandikci) [reply docs/modules/ROOT/pages/cpp-client.adoc:90]: Done, thanks!

File: docs/modules/ROOT/pages/cpp-client.adoc [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+<4> Now, we create the client with `config`.
```
[the remark this answers is not in the document]
Comment (yunussandikci) [reply docs/modules/ROOT/pages/cpp-client.adoc:92]: Done, thanks!

File: docs/modules/ROOT/pages/cpp-client.adoc [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
++
+image:cpp-client-dialog.png[Instructions for installing a .Net client]
+
+. Click on the C++ tab (the Java tab is active by default) and download the ZIP file as instructed in Step 1 of the dialog box.
+
+. Extract the ZIP file and run the command we provided, in the extracted folder, shown in Step 2.
+
+* Windows users may need update this command according to configuration of their system.  
+** If your target triple doesn't match with operation system, you may need to install Client with providing target suffix. For example: `vcpkg install "hazelcast-cpp-client[openssl]:x64-windows"`
+** Location of the compiled binary may change according to configuration and system defaults.
```
[the remark this answers is not in the document]
Comment (yunussandikci) [reply docs/modules/ROOT/pages/cpp-client.adoc:24]: Yes I tried this on both macOS and Windows and compiled binary was in different direction. It applies only to Windows users.

File: docs/modules/ROOT/pages/cpp-client.adoc [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+. When the cluster's state becomes "Running", the previously inactive button `Configure Client` on this page becomes active. Click on it and the following dialog box appears:
++
+image:cpp-client-dialog.png[Instructions for installing a .Net client]
+
+. Click on the C++ tab (the Java tab is active by default) and download the ZIP file as instructed in Step 1 of the dialog box.
+
+. Extract the ZIP file and run the command we provided, in the extracted folder, shown in Step 2.
+
+* Windows users may need update this command according to configuration of their system.  
+** If your target triple doesn't match with operation system, you may need to install Client with providing target suffix. For example: `vcpkg install "hazelcast-cpp-client[openssl]:x64-windows"`
```
[the remark this answers is not in the document]
Comment (yunussandikci) [reply docs/modules/ROOT/pages/cpp-client.adoc:23]: Typo, it should be `target triplet` You can check what is it from here: https://wiki.osdev.org/Target_Triplet

[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/modules/ROOT/pages/cpp-client.adoc:23]: Ah cool, thanks 😄  In that case we can say
```suggestion
** If the link:https://wiki.osdev.org/Target_Triplet[target triplet] doesn't match your operation system, you may need to provide a target suffix to the `install` command. For example: `vcpkg install "hazelcast-cpp-client[openssl]:x64-windows"`
```

**Review by JakeSCahill — CHANGES_REQUESTED** [review]
Just one final suggestion regarding the typo in `target triple`.

**Review by JakeSCahill — APPROVED** [review]
Thanks 😄

**Comment by yunussandikci** [comment]
Thanks @JakeSCahill I will merge it when we need it. (Release 3.0)

**Review by JakeSCahill — APPROVED** [review]

**Pull request merged**